server/users.go

```diff
@@ -191,17 +191,19 @@ func (s *Service) NewUser(w http.ResponseWriter, r *http.Request) {
 		Roles:    req.Roles,
 	}
 
-	// If req.SuperAdmin has been set, verify that it was done with the superadmin context.
-	// Even though req.SuperAdmin == true is logically equivalent to req.SuperAdmin it is
-	// more clear that this code should only be ran in the case that a user is trying to
-	// set the SuperAdmin field.
-	if req.SuperAdmin == true {
-		// Only allow users to set SuperAdmin if they have the superadmin context
-		if isSuperAdmin := hasSuperAdminContext(ctx); !isSuperAdmin {
-			Error(w, http.StatusBadRequest, "Cannot set SuperAdmin", s.Logger)
-			return
-		}
-		user.SuperAdmin = true
+	// Only allow users to set SuperAdmin if they have the superadmin context
+	// TODO(desa): Refactor this https://github.com/influxdata/chronograf/issues/2207
+	if isSuperAdmin := hasSuperAdminContext(ctx); isSuperAdmin {
```

Comment:
Would you change this to a function?  Because this is authorization related code, I would like to see code like this in one spot in case it needs to change in the future.  Now, this is debatable as you have tests that cover this, so, feel free to ignore

Reply:
Pulled out logic into a single function